contracts/src/v0.8/automation/dev/v2_3/AutomationRegistryBase2_3.sol

```diff
@@ -484,17 +503,22 @@ abstract contract AutomationRegistryBase2_3 is ConfirmedOwner {
     uint256 fastGasWei;
     uint256 linkUSD;
     uint256 nativeUSD;
+    BillingTokenPaymentParams billingToken;
     bool isTransaction;
   }
 
   /**
-   * @notice struct containing receipt information after a payment is made
-   * @member reimbursement the amount to reimburse a node for gas spent
-   * @member premium the premium charged to the user, shared between all nodes
+   * @notice struct containing receipt information about a payment or cost estimation
+   * @member gasChargeWei the amount to charge a user for gas spent
+   * @member premiumWei the premium charged to the user, shared between all nodes
+   * @member gasReimbursementJuels the amount to reimburse a node for gas spent
+   * @member premiumJuels the premium paid to NOPs, shared between all nodes
    */
   struct PaymentReceipt {
-    uint96 reimbursement;
-    uint96 premium;
+    uint96 gasChargeWei;
+    uint96 premiumWei;
+    uint96 gasReimbursementJuels;
+    uint96 premiumJuels;
```

Comment:
are premiumWei and premiumJuels the same USD amount in theory? we need to carry both of them in the receipt?

Reply:
yes, theoretically the same amount in USD. I'm including them both in the receipt because we need both values when we "handle payment" - the premiumWei gets billed to the user and the premiumJules gets paid to the NOP